From caa375e4361d717af1836c7e3d31a91067fb02c3 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Mon, 26 Mar 2018 11:16:36 +0300
Subject: [PATCH 3570/5242] MLK-17733: arm64: dts: fsl-imx8qm: Fix LVDS
 dual-channel panel

commit  2e9b537f3337fa0ef5da0dee88aef7aab57a4729 from
https://source.codeaurora.org/external/imx/linux-imx.git

The LVDS dual-channel pannel doesn't use the IT6263 LVDS-HDMI converter,
therefore we need to disable those nodes such that the IT6263 driver
won't disable the LDB nodes.
This is a special case, in which auto-detection cannot be used to detect
whether or not we have something connected on the LVDS ports.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 ...sl-imx8qm-lpddr4-arm2-jdi-wuxga-lvds1-panel.dts |   24 ++++++++++++++++++++
 .../fsl-imx8qm-mek-jdi-wuxga-lvds1-panel.dts       |   12 ++++++++++
 2 files changed, 36 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-jdi-wuxga-lvds1-panel.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-jdi-wuxga-lvds1-panel.dts
index e219c07..39bbdad 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-jdi-wuxga-lvds1-panel.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-jdi-wuxga-lvds1-panel.dts
@@ -27,6 +27,30 @@
 	};
 };
 
+&ldb1_phy {
+	status = "disabled";
+};
+
+&ldb1 {
+	status = "disabled";
+};
+
+&i2c1_lvds0 {
+	lvds-to-hdmi-bridge@4c {
+		status = "disabled";
+	};
+};
+
+&i2c1_lvds1 {
+	lvds-to-hdmi-bridge@4c {
+		status = "disabled";
+	};
+};
+
+&ldb2_phy {
+	status = "okay";
+};
+
 &ldb2 {
 	status = "okay";
 	fsl,dual-channel;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-jdi-wuxga-lvds1-panel.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-jdi-wuxga-lvds1-panel.dts
index 3dfb58a..8c4d450 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-jdi-wuxga-lvds1-panel.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-jdi-wuxga-lvds1-panel.dts
@@ -35,6 +35,18 @@
 	status = "disabled";
 };
 
+&i2c1_lvds0 {
+	lvds-to-hdmi-bridge@4c {
+		status = "disabled";
+	};
+};
+
+&i2c1_lvds1 {
+	lvds-to-hdmi-bridge@4c {
+	status = "disabled";
+	};
+};
+
 &ldb2_phy {
 	status = "okay";
 };
-- 
1.7.9.5

